sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorClientEventSource.cs

```diff
@@ -90,18 +94,20 @@ protected EventProcessorClientEventSource() : base(EventSourceName)
         /// <param name="identifier">A unique name used to identify the event processor.</param>
         /// <param name="eventHubName">The name of the Event Hub that the processor is associated with.</param>
         /// <param name="consumerGroup">The name of the consumer group that the processor is associated with.</param>
+        /// <param name="operationId">An identifier for the processing operation, allowing its activities to be correlated.</param>
         /// <param name="errorMessage">The message for the exception that occurred.</param>
         ///
-        [Event(22, Level = EventLevel.Error, Message = "An exception occurred while processing events for partition '{0}' by processor instance with identifier '{1}' for Event Hub: {2} and Consumer Group: {3}.  Error Message: '{4}'")]
+        [Event(22, Level = EventLevel.Error, Message = "An exception occurred while processing events for partition '{0}' by processor instance with identifier '{1}' for Event Hub: {2} and Consumer Group: {3}.  Operation Id: '{4}'; Error Message: '{5}'")]
         public virtual void EventBatchProcessingError(string partitionId,
                                                       string identifier,
                                                       string eventHubName,
                                                       string consumerGroup,
+                                                      string operationId,
```

Comment:
Is it safe to re-order the parameters here? Could this break consumers?

Reply:
I would think so.   Unlike error messages that are directly exposed to customers and parsed, logs are informational, and I don't believe we consider them a contract.

I wanted to keep the error message at the end of these because they're often long and sometimes  contain new lines that would make sniffing out the operation more difficult.   That said, I don't feel strongly, so if you do, I'll flip them around.

Reply:
Users can register their own listener and parse the individual arguments, so I think this would impact which argument represents what data. I would err on the side of leaving the order in place and just appending new params.

Reply:
By the way my comment is about the order of the arguments passed to WriteEvent, not about the actual format of the message. I think it is fair to change the message around.

Reply:
Chatted offline.  To ensure that we don't break structured use, the original ordering to the `WriteEvent` call was restored and the new parameters appended.    Log message was left in the desired format, unfortunately requiring out-of-order slugs.